src/structure_managers/property_block_sparse.hh

```diff
@@ -0,0 +1,600 @@
+          i_row++;
+        }
+        i_col++;
+      }
+      return feauture_row;
+    }
+
+    inline dense_t get_dense_rep() {
+      auto n_center{this->get_nb_item()};
+      keys_t all_keys{};
+      for (size_t i_center{0}; i_center < n_center; i_center++) {
+        auto keys = this->values[i_center].get_keys();
+        for (auto & key : keys) {
+          all_keys.insert(key);
+        }
+      }
+      dense_t features =
+          dense_t::Zero(this->get_nb_comp() * all_keys.size(), n_center);
+
+      for (size_t i_center{0}; i_center < n_center; i_center++) {
+        int i_feat{0};
+        for (const auto & key : all_keys) {
+          if (this->values[i_center].count(key) == 1) {
+            for (int i_pos{0}; i_pos < this->values[i_center][key].size();
+                 i_pos++) {
+              features(i_feat, i_center) = this->values[i_center][key](i_pos);
+              i_feat++;
+            }
+          }
+        }
+      }
+      return features;
+    }
+
+    //! Accessor for property by index for dynamically sized properties
+    inline decltype(auto) operator()(const size_t & index) {
+      return this->operator[](index);
+    }
+
+    template <size_t CallerLayer>
+    inline decltype(auto)
+    operator()(const ClusterRefKey<Order, CallerLayer> & id,
+               const key_t & key) {
+      static_assert(CallerLayer >= PropertyLayer,
+                    "You are trying to access a property that does not exist at"
+                    "this depth in the adaptor stack.");
+
+      return this->operator()(id.get_cluster_index(CallerLayer), key);
+    }
+
+    //! Accessor for property by index for dynamically sized properties
+    inline dense_ref_t operator()(const size_t & index, const key_t & key) {
+      return dense_ref_t(&this->values[index].at(key)(0, 0),
+                         this->values[index].at(key).rows(),
+                         this->values[index].at(key).cols());
+    }
+
+    //! getter to the underlying data storage
+    inline data_t & get_raw_data() { return this->values; }
+    //! get number of different distinct element in the property
+    //! (typically the number of center)
+    inline size_t get_nb_item() const { return this->values.size(); }
+
+    /**
+     * Accessor for last pushed entry for dynamically sized properties
+     */
+    inline decltype(auto) back() { return this->values.back(); }
+
+    //! push back data associated to a new center atom
+    inline void push_back(const input_data_t & ref) {
+      for (const auto & element : ref) {
+        const auto & value{element.second};
+        if (value.size() != this->get_nb_comp()) {
+          auto error{std::string("Size should match: ") +
+                     std::to_string(value.size()) + std::string(" != ") +
+                     std::to_string(this->get_nb_comp())};
+          throw std::length_error(error);
+        }
+      }
+
+      this->values.push_back(ref);
```

Comment:
in this vector `values`, which is a vector of the internal key map: I am a little confused. When it is a map, how can it be access later with `[]` with and index in `operator[]`?